### PR TITLE
[PATCH v3] Packet input timestamp testing improvements

### PR DESCRIPTION
**Pull request by MatiasElo**

### CODE REVIEW

File: test/validation/api/pktio/pktio.c [hunk 1]
```diff
@@ -3366,6 +3367,14 @@ static void pktio_test_pktin_ts(void)
 				       1, TXRX_MODE_SINGLE, ODP_TIME_SEC_IN_NS, false);
 		if (ret != 1)
 			break;
+
+		/* Compare to packet IO time to input timestamp */
+		pktio_time = odp_pktio_time(pktio_rx, NULL);
```
[the remark this answers is not in the document]
Comment (MatiasElo) [reply test/validation/api/pktio/pktio.c:3372]: Changed in V2.

File: test/validation/api/pktio/pktio.c [hunk 2]
```diff
@@ -3366,6 +3367,14 @@ static void pktio_test_pktin_ts(void)
...
+		CU_ASSERT_FATAL(odp_packet_has_ts(pkt_tbl[i]));
+		ts = odp_packet_ts(pkt_tbl[i]);
+		CU_ASSERT(odp_time_cmp(pktio_time, ts) >= 0);
+		CU_ASSERT(odp_time_diff_ns(pktio_time, ts) < 100 * ODP_TIME_MSEC_IN_NS);
```
[the remark this answers is not in the document]
Comment (MatiasElo) [reply test/validation/api/pktio/pktio.c:3376]: Print added in V2.

File: test/performance/odp_l2fwd.c [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /* Copyright (c) 2014-2018, Linaro Limited
```
[the remark this answers is not in the document]
Comment (MatiasElo) [reply test/performance/odp_l2fwd.c:1]: Commit message improved in V2.

File: example/packet/odp_packet_dump.c [hunk 4]
```diff
@@ -263,10 +265,17 @@ static int open_pktios(test_global_t *global)
...
+			return -1;
+		}
+
 		odp_pktio_print(pktio);
 
 		odp_pktio_config_init(&pktio_config);
-		pktio_config.pktin.bit.ts_all = 1;
+
+		if (pktio_capa.config.pktin.bit.ts_all)
+			pktio_config.pktin.bit.ts_all = 1;
```
[the remark this answers is not in the document]
Comment (MatiasElo) [reply example/packet/odp_packet_dump.c:278]: Changed in V2.

File: example/packet/odp_packet_dump.c [hunk 5]
```diff
@@ -501,6 +514,15 @@ static int print_packet(test_global_t *global, odp_packet_t pkt,
 
 	printf("PACKET [%" PRIu64 "]\n", num_packet);
 	printf("  time:            %" PRIu64 ".%09" PRIu64 " sec\n", sec, nsec);
+
+	if (input_ts_set) {
```
[the remark this answers is not in the document]
Comment (MatiasElo) [reply example/packet/odp_packet_dump.c:518]: Changed in V2.

**Review by TuomasTaipale — APPROVED** [review]

**Pull request merged**